src/gwt/src/org/rstudio/studio/client/common/spelling/TypoSpellChecker.java

```diff
@@ -208,6 +216,14 @@ void initialize(SpellingService spellingService, WorkbenchListManager workbenchL
          ).addCallback(loadSpellingWorker);
       }
 
+      if (domainSpecificWords_.isEmpty())
```

Comment:
Totally up to you, but feels like this could be simpler if the domain specific word list was an always-on dictionary instead of a hardcoded list shipped to the client and special-cased.

Reply:
Spent a little bit more thinking about it and I think for this specific use case and considering the overhead of additional dictionaries this is an adequate solution for now. I think if we get enough feature requests for it to be a living dictionary (and I can see this being super neat for users) we can do something fancy.

Reply:
👍